src/app/ui/shared/appoitment-form/appoitment-form.component.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { debounceTime, distinctUntilChanged, take } from 'rxjs/operators';
+import { DoctorDTO } from 'src/app/data/modelDTO/doctor-DTO';
+import { DoctorService } from 'src/app/services/doctor/doctor.service';
+import { DoctorServicesService } from 'src/app/services/doctor-service/doctor-services.service';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { UserDto } from 'src/app/data/modelDTO/user-dto';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-appoitment-form',
+  templateUrl: './appoitment-form.component.html',
+  styleUrls: ['./appoitment-form.component.scss'],
+})
+export class AppoitmentFormComponent implements OnInit {
+  sampleForm = new FormGroup({
+    startDate: new FormControl(null, Validators.required),
+    startTime: new FormControl(null, Validators.required),
+    medicId: new FormControl(null, Validators.required),
+    medicName: new FormControl(null, Validators.required),
+    medService: new FormControl(null, Validators.required),
+    medLocation: new FormControl(null, Validators.required),
+    pacientName: new FormControl('', Validators.required),
+    pacientId: new FormControl('', Validators.required),
+    pacientAnimal: new FormControl({}, Validators.required),
+  });
+  medicServices: string[];
+  formTitle: string;
+  pacientName: string;
+  users: Observable<any>;
+  animals: AnimalUtilInfo[];
+  doctorAppointment: DoctorsAppointmentDTO;
+  appoinementFormPlaceHolder;
+  @ViewChild('pacientList') pacientList: any;
+  @ViewChild('inputPacientName') inputPacientName: any;
+
+  constructor(
+    private doctorService: DoctorService,
+    private servicesService: DoctorServicesService,
+    private activeModal: NgbActiveModal,
+    private userService: UserService,
+    private doctorAppointmentService: DoctorAppointmentsService,
+    private appointmentFormService: AppointmentFormService
+  ) {}
+
+  ngOnInit(): void {
+    this.appoinementFormPlaceHolder = appointmentFormData;
+    this.formTitle = appointmentFormData.title;
+    this.pacientName = '';
+
+    this.doctorService
+      .getDoctorById('o2Jt7YS9zCWvBfDWY08X')
+      .pipe(take(1))
+      .subscribe((medic: DoctorDTO) => {
+        this.sampleForm.patchValue({
+          medicName: medic.doctorName,
+          medLocation: medic.location,
+          medicId: 'o2Jt7YS9zCWvBfDWY08X',
+        });
+      });
+
+    this.servicesService
+      .getAllServices('o2Jt7YS9zCWvBfDWY08X')
+      .pipe(take(1))
+      .subscribe((docServices: DoctorServiceDTO) => {
+        this.medicServices = docServices[0].services;
+      });
+  }
+
+  onSubmit(): void {
+    const newAnimalInfo = new AnimalUtilInfo()
+      .setName(this.sampleForm.value.pacientAnimal.animalName)
+      .setUid(this.sampleForm.value.pacientAnimal.animalId);
+
+    const newDoctorAppointment = new DoctorsAppointmentDTO()
```

Comment:
no validation here? why? - please validate the form

Reply:
added using reactive forms